### PR TITLE
EAPI 7: D/ROOT trailing slash

### PR DESCRIPTION
**Pull request by mgorny**

### CODE REVIEW

File: bin/ebuild-helpers/prepstrip [hunk 1]
```diff
@@ -143,7 +143,7 @@ save_elf_debug() {
 	local x=$1
 	local inode_debug=$2
 	local splitdebug=$3
-	local y=${ED}usr/lib/debug/${x:${#D}}.debug
+	local y=${ED%/}/usr/lib/debug/${x:${#D}}.debug
```
Comment (zmedico) [inline bin/ebuild-helpers/prepstrip]: We need to fix `${x:${#D}}` to account for `${D}` without a trailing slash, maybe use `local d=${D%/}` and `${#d}`.

File: bin/install-qa-check.d/80libraries [hunk 2]
```diff
@@ -6,7 +6,7 @@ scanelf_lib_check() {
 	# Check for shared libraries lacking SONAMEs
 	local qa_var="QA_SONAME_${ARCH/-/_}"
 	eval "[[ -n \${!qa_var} ]] && QA_SONAME=(\"\${${qa_var}[@]}\")"
-	f=$(scanelf -ByF '%S %p' "${ED}"{,usr/}lib*/lib*.so* | awk '$2 == "" { print }' | sed -e "s:^[[:space:]]${ED}:/:")
+	f=$(scanelf -ByF '%S %p' "${ED%/}"/{,usr/}lib*/lib*.so* | awk '$2 == "" { print }' | sed -e "s:^[[:space:]]${ED%/}:/:")
```
Comment (zmedico) [inline bin/install-qa-check.d/80libraries]: For the sed argument, it should be `${ED%/}/` in order to preserve behavior.

File: bin/install-qa-check.d/80libraries [hunk 3]
```diff
@@ -40,7 +40,7 @@ scanelf_lib_check() {
 	# Check for shared libraries lacking NEEDED entries
 	qa_var="QA_DT_NEEDED_${ARCH/-/_}"
 	eval "[[ -n \${!qa_var} ]] && QA_DT_NEEDED=(\"\${${qa_var}[@]}\")"
-	f=$(scanelf -ByF '%n %p' "${ED}"{,usr/}lib*/lib*.so* | awk '$2 == "" { print }' | sed -e "s:^[[:space:]]${ED}:/:")
+	f=$(scanelf -ByF '%n %p' "${ED%/}"/{,usr/}lib*/lib*.so* | awk '$2 == "" { print }' | sed -e "s:^[[:space:]]${ED%/}:/:")
```
Comment (zmedico) [inline bin/install-qa-check.d/80libraries]: `${ED%/}/` for sed argument again.

File: bin/ebuild-helpers/prepstrip (hunk 1 above)
Comment (mgorny) [reply bin/ebuild-helpers/prepstrip]: Ai, my grep didn't catch `${#D` etc.

**Comment by mgorny** [comment]
Done. I've named the var `d_noslash` to avoid confusion in the future.

**Pull request closed without merging**